crates/sui/tests/cli_tests.rs

```diff
@@ -1648,12 +1648,9 @@ async fn test_package_publish_nonexistent_dependency() -> Result<(), anyhow::Err
     .execute(context)
     .await;
 
-    let err = result.unwrap_err().to_string();
-    assert!(
-        err.contains("Dependency object does not exist or was deleted"),
-        "{}",
-        err
-    );
+    // publishing with a dependency that is not referenced in code is OK, as the dependency list
```

Comment:
It seems like in this case you need to update the test such that the dependency is actually referenced but does not exist, because that is what we're trying to test here.

Reply:
Yes, but now the issue will be that source verification will trigger an error that it cannot find the on-chain version of this dependency. FWIW, that Move package has set a random `published-at` field in the manifest.

Reply:
It's okay that the error wording is different, the point is that we're testing the error case here.